### ISSUE TITLE
When pressing AltGr+Q  (to get @) on a german keyboard layout no KeyTyped event is raised for Q. 

### ISSUE THREAD

**Issue by bj-rn** [TolikPylypchuk/SharpHook#47]
Hello again,

it appears when I opened [the last issue ](https://github.com/TolikPylypchuk/SharpHook/issues/45)I made some wrong assumptions and therefore looked in the wrong place.
The console output you posted pointed me in the right direction I think.

So when pressing (and releasing) `AltGr+Q` to get `@` the following events occur:

```
UioHookEvent: Type = KeyPressed; Time = 1683224417448; Mask = LeftCtrl; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683224417449; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683224417560; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcQ; RawCode = 81; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683224417616; Mask = RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683224417616; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683224417664; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcQ; RawCode = 81; KeyChar = ￿

```
As you can see no `KeyTyped` event is raised for `@`.

Second try was `AltGr+E`  ( `€`) also with a german layout, same behaviour:

```
UioHookEvent: Type = KeyPressed; Time = 1683225653859; Mask = LeftCtrl; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683225653859; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683225653955; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcE; RawCode = 69; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683225654019; Mask = RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683225654019; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683225654035; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcE; RawCode = 69; KeyChar = ￿
```

After that I tried `AltGr+ò` to get `@` with an [italien layout ](https://learn.microsoft.com/en-us/globalization/keyboards/kbdit) and it works:

```
UioHookEvent: Type = KeyPressed; Time = 1683226452760; Mask = LeftCtrl; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683226452760; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyPressed; Time = 1683226452881; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcBackquote; RawCode = 192; KeyChar = ￿
UioHookEvent: Type = KeyTyped; Time = 1683226452881; Mask = LeftCtrl, RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcUndefined; RawCode = 192; KeyChar = @
UioHookEvent: Type = KeyReleased; Time = 1683226452969; Mask = RightAlt; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcLeftControl; RawCode = 162; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683226452969; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcRightAlt; RawCode = 165; KeyChar = ￿
UioHookEvent: Type = KeyReleased; Time = 1683226452993; Mask = None; Reserved = None; Keyboard = KeyboardEventData: KeyCode = VcBackquote; RawCode = 192; KeyChar = ￿
```

Could you maybe try your sample app with a [german layout](https://learn.microsoft.com/en-us/globalization/keyboards/kbdgr) ?

TIA

**Comment by bj-rn** [comment]
Thanks for looking into it.

**Comment by ZeProgFactory** [comment]
I’ve a similar issue: 

![image](https://github.com/TolikPylypchuk/SharpHook/assets/17034030/9dd49551-8d21-4ee1-bfd1-76caec63d524)

When using English (United Kingdom) as language with a French Keyboard layout (same as physical configuration) everything is working fine. When I change to UK keyboard the `KeyTyped` event isn’t fired anymore.

**Comment by TolikPylypchuk** [comment]
You should create an issue in the [original libuiohook repository](https://github.com/kwhat/libuiohook). I don't think I can help here much.

**Comment by TolikPylypchuk** [comment]
Just a heads-up - I've fixed this issue in my own fork of libuiohook (I basically reimplemented getting characters for the `KeyTyped` events from scratch on Windows and it appears to work well - even for dead keys).

The fix is going to be available in the upcoming version 5.0.0.

**Comment by TolikPylypchuk** [comment]
I've just released version 5.0.0 which fixes this issue so it can finally be closed.
